src/dotty/tools/dotc/core/pickling/PickleFormat.scala

```diff
@@ -0,0 +1,454 @@
+package dotty.tools.dotc
+package core
+package pickling
+
+/************************************************************
+Notation:
+
+We use BNF notation. Terminal symbols start with at least two
+consecutive upper case letters. Each terminal is represented as a
+single byte tag. Non-terminals are mixed case. Prefixes of the form
+lower case letter*_ are for explanation of semantic content only, they
+can be dropped without changing the grammar.
+
+Micro-syntax:
+
+  LongNat       = Digit* StopDigit        // big endian, value fits in a Long without overflow
+  Nat           = LongNat                 // value fits in an Int without overflow
+  Digit         = 0 | ... | 127
+  StopDigit     = 128 | ... | 255         // value = digit - 128
+
+Macro-format:
+
+  File          = Header majorVersion_Nat minorVersion_Nat nameTable_Length Name* Section*
+  Header        = "5CA1AB1F"
+
+  Section       = NameRef Length Bytes
+  Length        = Nat                    // length of rest of entry in bytes
+
+  Name          = UTF8	        Length UTF8-CodePoint*
+                  QUALIFIED     Length qualified_NameRef selector_NameRef
+                  SIGNED        Length original_NameRef resultSig_NameRef paramSig_NameRef*
+                  EXPANDED      Length original_NameRef
+                  MODULECLASS   Length module_NameRef
+                  SUPERACCESSOR Length accessed_NameRef
+                  DEFAULTGETTER Length method_NameRef paramNumber_Nat
+                  ...
+
+  NameRef       = Nat                    // ordinal number of name in name table, starting from 1.
+
+Note: Unqualified names in the name table are strings. The context decides whether a name is
+a type-name or a term-name. The same string can represent both.
+
+Standard-Section: "ASTs" Tree*
+
+  Tree          = PACKAGE       Length Path Tree*
+                  Stat
+
+  Stat          = Term
+                  VALDEF        Length NameRef Type rhs_Tree Modifier*
+                  DEFDEF        Length NameRef TypeParam* Params* return_Type rhs_Tree
+                                       Modifier*
+                  TYPEDEF       Length NameRef (Type | Template) Modifier*
+                  IMPORT        Length qual_Term Selector*
```

Comment:
They would be, if there were not eliminated by FirstTransform, which runs before Pickling.